### PR TITLE
fix: Calling UI fixes ACC-143

### PR DESCRIPTION
**Pull request by MarcinR**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-143" title="ACC-143" target="_blank">StoryACC-143</a>  Context change on focus (3.2.1)
  </td></table>
  <br />
 

ACC-362 ACC-388 ACC-378 ACC-387 ACC-374 ACC-389 ACC-375
Update disabled mic icon and pageControl colors,
fix colors of calling action buttons after theme change
disable hiding bottom sheet without device orientation change
fix missing participants count label in bottom sheet when going back to calling UI
fix bottom sheet scrolling on horizontal and grid after rotation on iPad

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsInfoViewController/CallingActionsInfoViewController.swift [hunk 1]
```diff
@@ -72,6 +72,7 @@ class CallingActionsInfoViewController: UIViewController, UICollectionViewDelega
         self.participants = participants
         self.selfUser = selfUser
         super.init(nibName: nil, bundle: nil)
+        participantsHeaderLabel.text = L10n.Localizable.Call.Participants.showAll(participants.count).uppercased()
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsInfoViewController/CallingActionsInfoViewController.swift:75]: I have 2 questions:
1. Where did we set this title before?
2. Why can't we do it in the `setupViews` method? It doesn't looks good to do it in the `init()`

File: Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsView/CallingActionButton.swift [hunk 2]
```diff
@@ -41,7 +41,7 @@ class CallingActionButton: IconLabelButton {
         iconButton.setIconColor(SemanticColors.Button.iconCallingNormal, for: .normal)
         iconButton.setBackgroundImageColor(SemanticColors.Button.backgroundCallingNormal, for: .normal)
 
-        setTitleColor(SemanticColors.Button.textCallingNormal, for: . selected)
+        setTitleColor(SemanticColors.Button.textCallingNormal, for: .normal)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsView/CallingActionButton.swift:44]: You already set title color for `normal` state? Why didn't you set it for `selected`?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift [hunk 3]
```diff
@@ -32,20 +32,20 @@ class RoundedPageIndicator: RoundedBlurView {
...
-            pageControl.pageIndicatorTintColor = SemanticColors.View.backgroundDefaultWhite
+            backgroundColor = SemanticColors.View.borderInputBar
             blurView.isHidden = true
         }
 
         if #available(iOS 14.0, *) {
             pageControl.backgroundStyle = .minimal
             pageControl.allowsContinuousInteraction = false
         }
+        pageControl.isUserInteractionEnabled = false
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift:43]: Why?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift [hunk 4]
```diff
@@ -21,7 +21,7 @@ import UIKit
 import WireCommonComponents
 
 class RoundedPageIndicator: RoundedBlurView {
-    let pageControl = UIPageControl()
+    let pageControl = BorderedPageControl()
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift:24]: Why do we need custom UIPageControl? Shouldn't we change 
```
pageIndicatorTintColor
pageControl.currentPageIndicatorTintColor
backgroundColor
```
?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/Grid/GridView.swift [hunk 5]
```diff
@@ -223,15 +237,15 @@ extension GridView: UICollectionViewDelegateFlowLayout {
...
+            return DeveloperFlag.isUpdatedCallingUI ? 1.0 : .zero
+        }
 
     func collectionView(
         _ collectionView: UICollectionView,
         layout collectionViewLayout: UICollectionViewLayout,
         minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
 
-        return .zero
+            return DeveloperFlag.isUpdatedCallingUI ? 1.0 : .zero
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/CallGridView/Grid/GridView.swift:248]: Why do we need it?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/Grid/GridView.swift [hunk 6]
```diff
@@ -223,15 +237,15 @@ extension GridView: UICollectionViewDelegateFlowLayout {
         layout collectionViewLayout: UICollectionViewLayout,
         minimumLineSpacingForSectionAt section: Int) -> CGFloat {
 
-        return .zero
-    }
+            return DeveloperFlag.isUpdatedCallingUI ? 1.0 : .zero
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/CallGridView/Grid/GridView.swift:240]: Why do we need it?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/Grid/GridView.swift [hunk 7]
```diff
@@ -189,6 +190,19 @@ private extension GridView {
...
+            let itemsInRow = numberOfItemsIn(.row, for: indexPath)
+            let itemsInColumn = numberOfItemsIn(.column, for: indexPath)
+
+            let maxWidth = collectionView.bounds.size.width - CGFloat(itemsInRow - 1)
+            let maxHeight = collectionView.bounds.size.height - CGFloat(itemsInColumn - 1)
+
+            let width = maxWidth / CGFloat(itemsInRow)
+            let height = maxHeight / CGFloat(itemsInColumn)
+
+            return CGSize(width: width, height: height)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/CallGridView/Grid/GridView.swift:203]: Not easy to read and not super clear what and why we are doing here?

File: Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingBottomSheetViewController.swift [hunk 8]
```diff
@@ -86,14 +85,12 @@ class CallingBottomSheetViewController: BottomSheetContainerViewController {
...
-        if UIDevice.current.orientation.isLandscape {
-            let newConfiguration = BottomSheetConfiguration(height: view.bounds.height, initialOffset: bottomSheetMinimalOffset)
-            self.configuration = newConfiguration
-        } else {
-            let newConfiguration = BottomSheetConfiguration(height: bottomSheetMaxHeight, initialOffset: bottomSheetMinimalOffset)
-            self.configuration = newConfiguration
-        }
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        let height = (size.width > size.height) ? (size.height - headerBar.bounds.height) : bottomSheetMaxHeight
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingBottomSheetViewController.swift:90]: It's hard to read

File: Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingBottomSheetViewController.swift [hunk 9]
```diff
@@ -86,14 +85,12 @@ class CallingBottomSheetViewController: BottomSheetContainerViewController {
...
+        let newConfiguration = BottomSheetConfiguration(height: height, initialOffset: bottomSheetMinimalOffset)
+        guard self.configuration != newConfiguration else { return }
+        self.configuration = newConfiguration
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingBottomSheetViewController.swift:93]: Could you add some comments to explain what are you doing?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/BorderedPageControl.swift [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+import UIKit
+
+class BorderedPageControl: UIPageControl {
+    let selectedPageIndicator = UIImage.circle(size: CGSize(width: 12, height: 12), color: .accent(), filled: true)
+    let defaultPageIndicator = UIImage.circle(size: CGSize(width: 12, height: 12), color: .accent(), filled: false)
+
+    override var currentPage: Int {
+        didSet {
+            if #available(iOS 14.0, *) {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/CallGridView/BorderedPageControl.swift:27]: What will be for iOS 13?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/BorderedPageControl.swift [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+
+class BorderedPageControl: UIPageControl {
+    let selectedPageIndicator = UIImage.circle(size: CGSize(width: 12, height: 12), color: .accent(), filled: true)
+    let defaultPageIndicator = UIImage.circle(size: CGSize(width: 12, height: 12), color: .accent(), filled: false)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/CallGridView/BorderedPageControl.swift:23]: Why the both selectedPageIndicator and defaultPageIndicator have accent color?

File: Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsInfoViewController/CallingActionsInfoViewController.swift (hunk 1 above)
Comment (MarcinR) [reply Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsInfoViewController/CallingActionsInfoViewController.swift:75]: 1. We still set this in var participants{ didSet{}}, but when calling ui is back after minimalising calling ui it's not called.
2. done

File: Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsView/CallingActionButton.swift (hunk 2 above)
Comment (MarcinR) [reply Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsView/CallingActionButton.swift:44]: thats an error from automatic merging. Missed it

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift (hunk 3 above)
Comment (MarcinR) [reply Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift:43]: because we never handled taps on page control. And after tap selected dot is changing but grid is not scrolled

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift (hunk 4 above)
Comment (MarcinR) [reply Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift:24]: we need to clear all indicators to preferred image after page change

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/Grid/GridView.swift (hunk 6 above)
Comment (MarcinR) [reply Wire-iOS/Sources/UserInterface/Calling/CallGridView/Grid/GridView.swift:240]: to be sure that there will be space between cells. If we just make cells smaller then they will still be next to each other with empty scpace on bottom of the collection view. I used DeveloperFlag to prevent old tests from failing

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/Grid/GridView.swift (hunk 5 above)
Comment (MarcinR) [reply Wire-iOS/Sources/UserInterface/Calling/CallGridView/Grid/GridView.swift:248]: to be sure that there will be space between cells. If we just make cells smaller then they will still be next to each other with empty scpace on bottom of the collection view. I used DeveloperFlag to prevent old tests from failing

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/BorderedPageControl.swift (hunk 10 above)
Comment (MarcinR) [reply Wire-iOS/Sources/UserInterface/Calling/CallGridView/BorderedPageControl.swift:27]: ios13 already has circle borders instead of full dots 
image

File: Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingBottomSheetViewController.swift (hunk 9 above)
Comment (MarcinR) [reply Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingBottomSheetViewController.swift:93]: done

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 12]
```diff
@@ -1823,7 +1823,6 @@
 			dstSubfolderSpec = 10;
 			files = (
 				A96867AD26A5E59F00679D95 /* WireDataModel.xcframework in Embed Frameworks */,
-				01AAD75D294FD4CF00F1CFFD /* Datadog.xcframework in Embed Frameworks */,
```
Comment (KaterinaWire) [inline Wire-iOS.xcodeproj/project.pbxproj:1826]: Please revert these changes

File: Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsInfoViewController/CallingActionsInfoViewController.swift [hunk 13]
```diff
@@ -118,9 +118,10 @@ class CallingActionsInfoViewController: UIViewController, UICollectionViewDelega
         collectionView.bounces = true
         collectionView.delegate = self
         self.collectionView = collectionView
-            [actionsView, participantsHeaderView, collectionView].forEach(stackView.addArrangedSubview)
+        [actionsView, participantsHeaderView, collectionView].forEach(stackView.addArrangedSubview)
         CallParticipantsListCellConfiguration.prepare(collectionView)
         view.backgroundColor = SemanticColors.View.backgroundDefaultWhite
+        participantsHeaderLabel.text = L10n.Localizable.Call.Participants.showAll(participants.count).uppercased()
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsInfoViewController/CallingActionsInfoViewController.swift:124]: Let's make it more clean and move it to the Lina 115

File: Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsView/CallingActionButton.swift [hunk 14]
```diff
@@ -41,7 +41,6 @@ class CallingActionButton: IconLabelButton {
         iconButton.setIconColor(SemanticColors.Button.iconCallingNormal, for: .normal)
         iconButton.setBackgroundImageColor(SemanticColors.Button.backgroundCallingNormal, for: .normal)
 
-        setTitleColor(SemanticColors.Button.textCallingNormal, for: . selected)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsView/CallingActionButton.swift:44]: Why did you remove this code?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift (hunk 3 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift:50]: I think we should to handle it because we need it for Voice over

File: Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingBottomSheetViewController.swift [hunk 15]
```diff
@@ -86,14 +85,15 @@ class CallingBottomSheetViewController: BottomSheetContainerViewController {
...
-            let newConfiguration = BottomSheetConfiguration(height: view.bounds.height, initialOffset: bottomSheetMinimalOffset)
-            self.configuration = newConfiguration
-        } else {
-            let newConfiguration = BottomSheetConfiguration(height: bottomSheetMaxHeight, initialOffset: bottomSheetMinimalOffset)
-            self.configuration = newConfiguration
-        }
+    // after rotating device recalculate bottom sheet max height
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        let isHorizontal = size.width > size.height
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingBottomSheetViewController.swift:91]: Why don't use `UIDevice.current.orientation` ?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift [hunk 16]
```diff
@@ -31,21 +33,26 @@ class RoundedPageIndicator: RoundedBlurView {
 
         addSubview(pageControl)
         pageControl.currentPageIndicatorTintColor = .accent()
+        pageControl.pageIndicatorTintColor = SemanticColors.Switch.borderOffStateEnabled
+
         if DeveloperFlag.isUpdatedCallingUI {
-            pageControl.pageIndicatorTintColor = SemanticColors.View.backgroundDefaultWhite
+            backgroundColor = SemanticColors.View.borderInputBar
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift:39]: I didn't find that we use this `backgroundColor`. Do we use it?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift [hunk 17]
```diff
@@ -75,3 +91,32 @@ class RoundedPageIndicator: RoundedBlurView {
...
+    class func circle(filled: Bool) -> UIImage? {
+        let size = CGSize(width: 12, height: 12)
+        let lineWidth = 1.0
+
+        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)
+        guard let context = UIGraphicsGetCurrentContext() else {
+                return nil
+        }
+        if filled {
+            context.setFillColor(UIColor.black.cgColor)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift:106]: Why `black`?

File: Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift [hunk 18]
```diff
@@ -75,3 +91,32 @@ class RoundedPageIndicator: RoundedBlurView {
...
+        } else {
+            context.setStrokeColor(UIColor.black.cgColor)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Calling/CallGridView/RoundedPageIndicator.swift:108]: Why `black`?

**Review by KaterinaWire — APPROVED** [review]
